src/runner/basic.rs

```diff
@@ -121,6 +272,7 @@ type Failed = bool;
 pub struct Basic<
     World,
     F = WhichScenarioFn,
+    R = RetryOptionsFn,
```

Comment:
@ilslv maybe we could just `Box<dyn FnMut>` here instead of introducing yet another type parameter? Doesn't seem to be critical.